src/common/hooks/useZeroApprove.ts

```diff
@@ -4,18 +4,65 @@ import { useApproveCallback } from './useApproveCallback'
 import { Currency, CurrencyAmount } from '@uniswap/sdk-core'
 import { useSetAtom } from 'jotai'
 import { zeroApprovalState } from '../state/useZeroApprovalState'
+import { useSafeApiKit } from 'api/gnosisSafe/hooks/useSafeApiKit'
+import { SafeMultisigTransactionResponse } from '@safe-global/safe-core-sdk-types'
+import { useIsActiveWallet } from 'legacy/hooks/useIsActiveWallet'
+import { walletConnectConnection } from 'modules/wallet/web3-react/connection/walletConnect'
+import { useIsSafeWallet } from 'modules/wallet'
+
+function wait(ms = 1000) {
+  return new Promise((resolve) => {
+    setTimeout(resolve, ms)
+  })
+}
```

Comment:
duplicated, you have it here https://github.com/cowprotocol/cowswap/blob/fix-approval-edge-case/src/legacy/utils/retry.ts#L1